src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,70 @@
+package ecommerce.controller
+
+import ecommerce.model.Product
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.test.annotation.DirtiesContext
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class ProductControllerTest {
+    @Test
+    fun createProduct() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(Product(id = 100, name = "test", price = 20.0, img = "img1", 2))
+                .contentType(ContentType.JSON)
+                .`when`().post("/products")
+                .then().log().all().extract()
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value())
```

Comment:
```suggestion
                .`when`().post("/products")
                .then().log().all()
                .assertThat().statusCode(HttpStatus.CREATED.value())
```
When you have the time, I recommend reading the official documentation of the libraries that we use. `RestAssured` itself offers a lot of asserting tools!

This is just a comment to show that there are so many cool things we don't know about the things that we use, but in practice, there are often times when we add bunch of dependencies only to find out that it could've been done with only one : _the one we already had for several years_ 😂

Reply:
Also, I think there should be more assertions about `updating the product information`. Checking only the http status code doesn't seem enough for me 🤔 . Why don't we add more assertions into the tests?